### PR TITLE
Enable the serialization of sil_vtables by default and completely remove the -sil-serialize-vtables option

### PR DESCRIPTION
**Pull request by swiftix**
Only sil_vtables of public classes with fixed layout are serialized.

While removing `-sil-serialize-vtables`, also remove all remaining mentions of `-sil-serialize-all` in CMake files and builds scripts

### CODE REVIEW

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

File: stdlib/CMakeLists.txt [hunk 1]
```diff
@@ -93,15 +93,11 @@ add_custom_target(swift-test-stdlib ALL
 
 if(SWIFT_STDLIB_ENABLE_RESILIENCE)
   set(STDLIB_SIL_SERIALIZE_ALL)
```
Comment (slavapestov) [inline stdlib/CMakeLists.txt]: You can remove all mentions of STDLIB_SIL_SERIALIZE_ALL from the CMake scripts now, and also remove the flag from build-script-impl

**Comment by slavapestov** [comment]
Awesome!

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please smoke test

**Comment by swiftix** [comment]
@swift-ci please test

**Comment by swiftix** [comment]
@swift-ci please test

File: CMakeLists.txt [hunk 2]
```diff
@@ -296,20 +292,11 @@ option(SWIFT_STDLIB_ENABLE_SIL_OWNERSHIP
   "Build the standard libraries and overlays with sil ownership enabled."
   FALSE)
 
-option(SWIFT_STDLIB_SIL_SERIALIZE_ALL
-    "Build the standard libraries and overlays serializing all method bodies"
-    TRUE)
-
 if(SWIFT_SERIALIZE_STDLIB_UNITTEST AND SWIFT_STDLIB_ENABLE_RESILIENCE)
```
Comment (slavapestov) [inline CMakeLists.txt]: I think SWIFT_SERIALIZE_STDLIB_UNITTEST  isn't needed anymore either

File: include/swift/SIL/SILVTable.h [hunk 3]
```diff
@@ -93,14 +94,18 @@ class SILVTable : public llvm::ilist_node<SILVTable>,
   /// The ClassDecl mapped to this VTable.
   ClassDecl *Class;
 
+  /// Whether or not this vtable is serialized, which allows
+  /// devirtualization from another module.
+  bool Serialized;
```
Comment (slavapestov) [inline include/swift/SIL/SILVTable.h]: If you want to be cute, make this a : 1 and NumEntries : 31 :-)

File: utils/build-script-impl [hunk 4]
```diff
@@ -172,7 +171,6 @@ KNOWN_SETTINGS=(
     build-swift-static-sdk-overlay    ""         "set to 1 to build static variants of the Swift SDK overlay"
     build-swift-examples        "1"              "set to 1 to build examples"
     build-swift-remote-mirror   "1"              "set to 1 to build the Swift Remote Mirror library"
-    build-serialized-stdlib-unittest "0"         "set to 1 to build the StdlibUnittest module with -sil-serialize-all"
```
Comment (slavapestov) [inline utils/build-script-impl]: This is also mentioned in presets.ini I think

**Comment by swiftix** [comment]
@swift-ci please test

**Comment by swiftix** [comment]
@swift-ci please test

**Comment by swiftix** [comment]
@swift-ci please smoke benchmark

**Comment by swiftix** [comment]
@swift-ci please smoke benchmark

**Comment by swiftix** [comment]
@swift-ci please smoke test compiler performance

**Comment by swiftix** [comment]
@swift-ci please smoke test compiler performance

**Comment by swiftix** [comment]
@swift-ci Please test source compatibility

**Comment by swiftix** [comment]
@swift-ci Please test source compatibility

**Comment by swiftix** [comment]
@slavapestov I made the changes you proposed in the comments. Will merge once all CI tests are finished.

**Pull request merged**